onnxruntime/core/providers/trt/trt_execution_provider.cc

```diff
@@ -0,0 +1,420 @@
+      }
+
+      input_binding_indexes.push_back(bindingIndex);
+      input_dim_sizes.push_back(dim_size);
+    }
+
+    // Get output shape and binding index
+    output_shapes.resize(trt_network->getNbOutputs());
+    for (int i = 0, end = trt_network->getNbOutputs(); i < end; ++i) {
+      const std::string& name = trt_network->getOutput(i)->getName();
+      size_t bindingIndex = trt_engine->getBindingIndex(name.c_str());
+
+      nvinfer1::Dims4 dimensions = static_cast<nvinfer1::Dims4&&>(trt_engine->getBindingDimensions(static_cast<int>(bindingIndex)));
+      size_t dim_size = 1;
+      for (int j = 0, end = dimensions.nbDims; j < end; ++j) {
+        output_shapes[i].push_back(dimensions.d[j]);
+        dim_size *= dimensions.d[j];
+      }
+
+      output_binding_indexes.push_back(bindingIndex);
+      output_dim_sizes.push_back(dim_size);
+    }
+
+    int num_inputs = input_binding_indexes.size();
+    int num_outputs = output_binding_indexes.size();
+    ORT_ENFORCE(trt_engine->getNbBindings() == (num_inputs + num_outputs));
+
+    // Save engine, context and input/output info to map
+    parsers_.emplace(fused_node->Name(), std::move(trt_parser));
+    engines_.emplace(fused_node->Name(), std::move(trt_engine));
+    contexts_.emplace(fused_node->Name(), std::move(trt_context));
+    input_info_[fused_node->Name()].push_back(input_indexes);
+    input_info_[fused_node->Name()].push_back(input_binding_indexes);
+    input_info_[fused_node->Name()].push_back(input_dim_sizes);
+    output_info_[fused_node->Name()].push_back(output_binding_indexes);
+    output_info_[fused_node->Name()].push_back(output_dim_sizes);
+    output_shapes_[fused_node->Name()] = output_shapes;
+
+    // Create function state
+    NodeComputeInfo compute_info;
+    compute_info.create_state_func = [=](ComputeContext* context, FunctionState* state) {
+      std::unique_ptr<TRTFuncState> p = std::make_unique<TRTFuncState>();
+      *p = {context->allocate_func, context->release_func, context->allocator_handle, parsers_[context->node_name].get(), engines_[context->node_name].get(), contexts_[context->node_name].get(),
+            input_info_[context->node_name], output_info_[context->node_name], output_shapes_[context->node_name]};
+      *state = p.release();
+      return 0;
+    };
+
+    // Release function state
+    compute_info.release_state_func = [](FunctionState state) {
+      if (state)
+        delete static_cast<TRTFuncState*>(state);
+    };
+
+    // Create compute function
+    compute_info.compute_func = [](FunctionState state, ONNXRunTimeTensor* input_tensors, size_t num_inputs, ONNXRunTimeTensor* output_tensors, size_t num_outputs) {
+      ORT_UNUSED_PARAMETER(num_inputs);
+      ORT_UNUSED_PARAMETER(num_outputs);
+      TRTFuncState* trt_state = reinterpret_cast<TRTFuncState*>(state);
+      std::vector<int> input_indexes = (trt_state->input_info)[0];
+      std::vector<int> input_binding_indexes = (trt_state->input_info)[1];
```

Comment:
Is the intention to copy all these values from the state, or could you just take a reference to them?

Reply:
use reference instead. Thanks.